### PR TITLE
added sequence clustering to pdb dataset

### PR DESCRIPTION
**Pull request by kierandidi**
Exposed pdb manager functionality to cluster sequences to the pdb dataset so that splits can be created based on sequence similarity.

### CODE REVIEW

File: proteinworkshop/datasets/pdb_dataset.py [hunk 1]
```diff
@@ -44,11 +46,14 @@ def __init__(
         self.remove_pdb_unavailable = remove_pdb_unavailable
         self.min_length = min_length
         self.max_length = max_length
-        self.split_sizes = split_sizes
+        self.split_ratios = split_ratios
```
Comment (a-r-j) [inline proteinworkshop/datasets/pdb_dataset.py:49]: Can we add a check these sum to 1

File: proteinworkshop/datasets/pdb_dataset.py [hunk 2]
```diff
@@ -29,7 +29,9 @@ def __init__(
         remove_ligands: List[str],
         remove_non_standard_residues: bool,
         remove_pdb_unavailable: bool,
-        split_sizes: List[float],
+        split_ratios: List[float],
+        split_type: str,
```
Comment (a-r-j) [inline proteinworkshop/datasets/pdb_dataset.py:33]: Can we use a `Literal`

File: proteinworkshop/datasets/pdb_dataset.py [hunk 3]
```diff
@@ -109,13 +114,20 @@ def create_dataset(self):
...
+            splits = pdb_manager.split_df_proportionally(
+                df=pdb_manager.df,
+                splits=self.splits,
+                split_ratios=self.split_ratios,
+            )
+        
+        elif self.split_type == "sequence_similarity":
+            log.info(f"Splitting dataset via sequence-similarity split into {self.split_ratios}...")
+            log.info(f"Using {self.split_sequence_similarity} sequence similarity for split")
+            pdb_manager.cluster(min_seq_id=self.split_sequence_similarity, update=True, overwrite=True)
```
Comment (a-r-j) [inline proteinworkshop/datasets/pdb_dataset.py:128]: What are your thoughts on having `overwrite=True` by default. Seems like something we'd like to control? Will the split be deterministic?

File: proteinworkshop/datasets/pdb_dataset.py [hunk 4]
```diff
@@ -29,7 +29,9 @@ def __init__(
         remove_ligands: List[str],
         remove_non_standard_residues: bool,
         remove_pdb_unavailable: bool,
-        split_sizes: List[float],
+        split_ratios: List[float],
```
Comment (a-r-j) [inline proteinworkshop/datasets/pdb_dataset.py:32]: Can we follow the same convention (`train_val_test`) https://github.com/a-r-j/ProteinWorkshop/blob/716a1b2e9f5115950946f8d980fc4800b4bf167c/proteinworkshop/datasets/igfold.py#L53

File: proteinworkshop/datasets/pdb_dataset.py (hunk 1 above)
Comment (kierandidi) [reply proteinworkshop/datasets/pdb_dataset.py:50]: done

File: proteinworkshop/datasets/pdb_dataset.py (hunk 2 above)
Comment (kierandidi) [reply proteinworkshop/datasets/pdb_dataset.py]: done

File: proteinworkshop/datasets/pdb_dataset.py (hunk 3 above)
Comment (kierandidi) [reply proteinworkshop/datasets/pdb_dataset.py]: good observation; tried it out and splits are not deterministic. Changed it to be False by default now

Comment (kierandidi) [reply proteinworkshop/datasets/pdb_dataset.py]: the case of different seq id levels is already covered so that is still possible

Comment (a-r-j) [reply proteinworkshop/datasets/pdb_dataset.py]: I think it makes sense to add an `overwrite` arg to the datamodule which is passed down to the `create_dataset` call. A few datamodules already have this functionality.

File: proteinworkshop/datasets/pdb_dataset.py (hunk 4 above)
Comment (kierandidi) [reply proteinworkshop/datasets/pdb_dataset.py]: done

File: proteinworkshop/datasets/pdb_dataset.py [hunk 5]
```diff
@@ -44,11 +46,15 @@ def __init__(
...
+        assert sum(train_val_test) == 1, f"train_val_test need to sum to 1, but sum to {sum(train_val_test)}"
+        self.train_val_test = train_val_test
+        self.split_type = split_type
+        self.split_sequence_similarity = split_sequence_similiarity
+        self.splits = ["train", "val", "test"]
 
     def create_dataset(self):
         log.info(f"Initializing PDBManager in {self.path}...")
-        pdb_manager = PDBManager(root_dir=self.path)
+        pdb_manager = PDBManager(root_dir=self.path, splits=self.splits, train_val_test=self.train_val_test)
```
Comment (a-r-j) [inline proteinworkshop/datasets/pdb_dataset.py:59]: careful, I think the PDBManager arg is still `split_ratios`

File: proteinworkshop/datasets/pdb_dataset.py (hunk 3 above)
Comment (kierandidi) [reply proteinworkshop/datasets/pdb_dataset.py]: done, lmk what you think

Comment (a-r-j) [reply proteinworkshop/datasets/pdb_dataset.py]: Nice! LGTM

File: proteinworkshop/datasets/pdb_dataset.py (hunk 5 above)
Comment (amorehead) [reply proteinworkshop/datasets/pdb_dataset.py:59]: Agreed

**Comment by kierandidi** [comment]
Tested runthrough that works now, both with and without overwrite

**Pull request merged**